huaweicloud/services/acceptance/rf/resource_huaweicloud_rf_stack_test.go

```diff
@@ -0,0 +1,605 @@
+					resource.TestCheckResourceAttrSet(rName, "created_at"),
+					resource.TestCheckResourceAttrSet(rName, "updated_at"),
+				),
+			},
+			{
+				Config: testAccStack_withUri_step1(name, basicTemplateInJsonFormat(name)),
+				Check: resource.ComposeTestCheckFunc(
+					rc.CheckResourceExists(),
+				),
+			},
+			{
+				Config: testAccStack_withUri_step2(name, basicTemplateInJsonFormat(name), variableContent),
+				Check: resource.ComposeTestCheckFunc(
+					rc.CheckResourceExists(),
+				),
+			},
+			{
+				ResourceName:      rName,
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateVerifyIgnore: []string{
+					"agency",
+					"template_uri",
+					"vars_uri",
+				},
+			},
+		},
+	})
+}
+
+func testAccStack_withUri_base(name, template, vars string) string {
+	return fmt.Sprintf(`
+
+resource "huaweicloud_obs_bucket" "test" {
+  bucket = "%[1]s"
+  acl    = "private"
+}
+
+resource "huaweicloud_obs_bucket_policy" "test" {
+  bucket = huaweicloud_obs_bucket.test.bucket
+  policy = <<EOT
+{
+  "Statement": [
+    {
+      "Sid": "RF-Access",
+      "Effect": "Allow",
+      "Principal": {
+        "ID": ["*"]
+      },
+      "Action": [
+        "GetObject"
+      ],
+      "Resource": [
+        "${huaweicloud_obs_bucket.test.bucket}/rf/resource_stack/uri_test/*"
+      ]
+    }
+  ]
+}
+EOT
+}
+
+resource "huaweicloud_obs_bucket_object" "template" {
+  bucket       = huaweicloud_obs_bucket.test.bucket
+  key          = "rf/resource_stack/uri_test/template.tf.json"
+  content_type = "application/json"
+  content      = %[2]s
+}
+
+resource "huaweicloud_obs_bucket_object" "variable" {
+  bucket       = huaweicloud_obs_bucket.test.bucket
+  key          = "rf/resource_stack/uri_test/resource.tfvars"
+  content_type = "application/octet-stream"
+  content      = %[3]s
+}
+`, name, template, vars)
+}
+
+func testAccStack_withUri_step1(name, template string) string {
+	return fmt.Sprintf(`
+%[1]s
+
+resource "huaweicloud_rf_stack" "test" {
+  name = "%[2]s"
+
+  agency {
+    name          = "rf_admin_trust" // System RF agency
+    provider_name = "huaweicloud"
+  }
+
+  template_uri = "https://${huaweicloud_obs_bucket.test.bucket_domain_name}/${huaweicloud_obs_bucket_object.template.id}"
+}
+`, testAccStack_withUri_base(name, template, variableContent), name, template)
+}
+
+func testAccStack_withUri_step2(name, template, vars string) string {
+	return fmt.Sprintf(`
+%[1]s
+
+resource "huaweicloud_rf_stack" "test" {
+  name = "%[2]s"
+
+  agency {
+    name          = "rf_admin_trust" // System RF agency
+    provider_name = "huaweicloud"
+  }
+
+  template_uri = "https://${huaweicloud_obs_bucket.test.bucket_domain_name}/${huaweicloud_obs_bucket_object.template.id}"
+  vars_uri     = "https://${huaweicloud_obs_bucket.test.bucket_domain_name}/${huaweicloud_obs_bucket_object.variable.id}"
+}
+`, testAccStack_withUri_base(name, template, variableContent), name, template, vars)
```

Comment:
the` last two params template and vars are not used

Reply:
fixed.